### PR TITLE
Make meta a class

### PR DESCRIPTION
**Pull request by sergiusens**
This is work to enable easier handling of the global `common.get_Xdir()` calls 

LP: #1551849

### CODE REVIEW

File: snapcraft/internal/meta.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
```
Comment (kyrofa) [inline snapcraft/internal/meta.py]: Should this be `_meta.py`?

File: snapcraft/internal/meta.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    :return: meta_dir.
+    """
+    packaging = _SnapPackaging(config_data, snap_dir, parts_dir)
+    packaging.write_snap_yaml()
+    packaging.setup_assets()
+
+    return packaging.meta_dir
+
+
+class _SnapPackaging:
```
Comment (kyrofa) [inline snapcraft/internal/meta.py]: I'm not sure I completely understand the benefit of having a class do this versus the standalone functions. Am I missing something?

File: snapcraft/internal/meta.py (hunk 1 above)
Comment (sergiusens) [inline snapcraft/internal/meta.py]: It is just like `states` (although a package); as long as it is in `internal` it should be fine.

File: snapcraft/internal/meta.py (hunk 2 above)
Comment (sergiusens) [inline snapcraft/internal/meta.py]: I don't need to pass around `snap_dir` and `parts_dir` which if `common.get` weren't used all over would really complicate the API.

Comment (kyrofa) [inline snapcraft/internal/meta.py]: Ah, okay, I gotcha.

Comment (sergiusens) [inline snapcraft/internal/meta.py]: I also plan to get rid of `common.assemble_env` this could be done in a much better way.

**Comment by kyrofa** [comment]
Looks good! :+1:

**Pull request merged**